rust/src/cli/service.rs

```diff
@@ -99,3 +111,86 @@ fn relocate_file(file_path: &Path) -> Result<(), CliError> {
     );
     Ok(())
 }
+
+fn pin_iface_name(cfg_dir: &Path) -> Result<(), CliError> {
+    let file_path = cfg_dir.join(PIN_STATE_FILENAME);
+    let mut fd = std::fs::File::open(&file_path)?;
+    let mut content = String::new();
+    fd.read_to_string(&mut content)?;
+    let pin_state: NetworkState = serde_yaml::from_str(&content)?;
+    let mut cur_state = NetworkState::new();
+    cur_state.set_kernel_only(true);
+    cur_state.set_running_config_only(true);
+    cur_state.retrieve()?;
+
+    for cur_iface in cur_state
+        .interfaces
+        .iter()
+        .filter(|i| i.iface_type() == InterfaceType::Ethernet)
+    {
+        let cur_mac = match cur_iface.base_iface().mac_address.as_ref() {
+            Some(c) => c,
+            None => continue,
+        };
+        if pin_state
+            .interfaces
+            .get_iface(cur_iface.name(), cur_iface.iface_type())
+            .is_none()
+        {
+            for pin_iface in pin_state
+                .interfaces
+                .iter()
+                .filter(|i| i.iface_type() == InterfaceType::Ethernet)
+            {
+                if pin_iface.base_iface().mac_address.as_ref() == Some(cur_mac)
+                    && pin_iface.name() != cur_iface.name()
+                {
+                    log::info!(
+                        "Pining the interface with MAC {cur_mac} to \
+                        interface name {}",
+                        pin_iface.name()
+                    );
+                    pin_iface_name_via_systemd_link(cur_mac, pin_iface.name())?;
+                }
+            }
+        }
+    }
+
+    relocate_file(&file_path)?;
+    Ok(())
+}
+
+fn pin_iface_name_via_systemd_link(
+    mac: &str,
+    iface_name: &str,
+) -> Result<(), CliError> {
+    let link_dir = Path::new(SYSTEMD_NETWORK_LINK_FOLDER);
+    if !link_dir.exists() {
+        std::fs::create_dir(&link_dir)?;
+    }
+
+    let content =
+        format!("[Match]\nMACAddress={mac}\n\n[Link]\nName={iface_name}\n");
+
+    // 98 here is important as it should be invoked after others but before
+    // 99-default.link
+    let file_path = link_dir.join(format!("98-{iface_name}.link"));
+
+    let mut fd = std::fs::OpenOptions::new()
+        .write(true)
+        .truncate(true)
+        .create(true)
+        .open(&file_path)
+        .map_err(|e| {
+            CliError::from(format!(
+                "Failed to store captured states to file {}: {e}",
+                file_path.display()
+            ))
+        })?;
+    fd.write_all(content.as_bytes())?;
```

Comment:
I think this is all just:
```
    std::fs::write(&file_path, content.as_bytes()).map_err(|e| ...)?;
```

Reply:
it does not override, I guess. Just copy from my code bank.